kernel/src/actions/crc.rs

```diff
@@ -94,14 +93,41 @@ pub(crate) struct DeletedRecordCountsHistogram {
     pub(crate) deleted_record_counts: Vec<i64>,
 }
 
+/// Attempt to read Protocol and Metadata from a CRC file. This will return an error if
+/// protocol/metadata cannot be found in the CRC file (specification requires both to be
+/// present).
+pub(crate) fn try_protocol_metadata_from_crc(
+    path: FileMeta,
+    engine: &dyn Engine,
+) -> DeltaResult<(Protocol, Metadata)> {
+    let json_handler = engine.json_handler();
+    let crc_schema = Crc::to_schema().into();
```

Comment:
nit: should we wrap this in a `LazyLock`?

Reply:
yea I actually think we should make the `derive(schema)` do something like internally generate a static `Struct::SCHEMA`?

Reply:
but will probably take in a follow up: https://github.com/delta-io/delta-kernel-rs/issues/1145